go/enclave/db/storage.go

```diff
@@ -213,25 +207,19 @@ func (s *storageImpl) FetchLogs(blockHash common.L1RootHash) ([]*types.Log, erro
 	return logs, nil
 }
 
-// TODO - #718 - This method has behaviour that's too dependent on various flags. Decompose.
-func (s *storageImpl) StoreNewHeads(l1Head common.L1RootHash, rollup *core.Rollup, receipts []*types.Receipt, isNewRollup bool, isNewL1Block bool) error {
+func (s *storageImpl) UpdateL2HeadForL1Block(l1Head common.L1RootHash, l2Head *core.Rollup, receipts []*types.Receipt) error {
```

Comment:
This method now only makes changes related to modifying which rollup a given L1 block is associated with.